nova/virt/vmwareapi/session.py

```diff
@@ -15,12 +15,54 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
+import abc
+import itertools
+import six
+
+from oslo_log import log as logging
+from oslo_utils import excutils
 from oslo_vmware import api
+from oslo_vmware import exceptions as vexc
 from oslo_vmware import vim
+from oslo_vmware.vim_util import get_moref_value
 
 import nova.conf
 
 CONF = nova.conf.CONF
+LOG = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class StableMoRefProxy(object):
+    """Abstract Basis class which acts as a proxy
+    for Managed-Object-References (MoRef).
+    Those references are usually "stable", meaning
+    they don't change over the life-time of the object.
+
+    But usually doesn't mean always. In that case, we
+    need to fetch the reference again via some search method,
+    which uses a guaranteed stable identifier (names, uuids, ...)
+    """
+    def __init__(self, ref):
+        self.moref = ref
+
+    @property
+    def __class__(self):
+        # Suds uses instanceof() to determine the serialiser,
+        # instead of duck-typing, so we have to fake it
+        return self.moref.__class__
```

Comment:
Do you mean `isinstance()`? I could not find `instanceof()` in the suds code. Also, does this work? Testing it with a simple example suggests that one cannot overwrite `__class__` with a property:
```python
class A():
    def __init__(self, moref):
        self.moref = moref

    @property
    def __class__(self):
        return self.moref.__class__

    def __getattr__(self, name):
        return getattr(self.moref, name)


a = A(1)
print(a.__class__)
print(getattr(1, '__class__'))
print(a.moref.__class__)
print(isinstance(a, A))
print(isinstance(a, int))
```
```
$ python /tmp/bla.py 
__main__.A
<type 'int'>
<type 'int'>
True
False
```

Reply:
Yes, that comment is incorrect. I use `isinstance()` myself in `_call_method` to determine that an argument is derived of `StableMoRefProxy`. It wouldn't work, if what I stated in the comment would be true.

It rather uses  `obj.__class__.__name__`  to determine the text of the tag. I'll fix the comment.